webextension/aboutConfigPrefs.js

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+/* global ExtensionAPI, Services */
+
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+ChromeUtils.import("resource://gre/modules/ExtensionCommon.jsm");
+ChromeUtils.defineModuleGetter(this, "AddonManager", "resource://gre/modules/AddonManager.jsm");
+
+const EventManager = ExtensionCommon.EventManager;
+
+this.aboutConfigPrefs = class extends ExtensionAPI {
+  getAPI(context) {
+    const extensionIDBase = context.extension.id.split("@")[0];
+    const prefBranchPrefix = `extensions.${extensionIDBase}.`;
+    const prefBranch = Services.prefs.getBranch(prefBranchPrefix);
+    const prefChangeEventName = "experiments.aboutConfigPrefs.onPrefChange";
+
+    function get(type, name) {
+      try {
+        return prefBranch[`get${type}Pref`](name);
+      } catch (_) {
+        return undefined;
+      }
+    }
+
+    const prefsToClearOnUninstall = new Set();
+
+    return {
+      experiments: {
+        aboutConfigPrefs: {
+          onPrefChange: new EventManager(context, prefChangeEventName, (fire, name) => {
+            const callback = () => {
+              fire.async(name);
+            };
+            Services.prefs.addObserver(`${prefBranchPrefix}${name}`, callback);
+            return () => {
+              Services.prefs.removeObserver(`${prefBranchPrefix}${name}`, callback);
+            };
+          }).api(),
+          async clearPref(name) {
+            Services.prefs.clearUserPref(`${prefBranchPrefix}${name}`);
+          },
+          async clearPrefsOnUninstall(names) {
+            if (!names.length) {
+              return;
+            }
+            if (!prefsToClearOnUninstall.size) {
+              AddonManager.addAddonListener({
```

Comment:
What you're doing here works, though extension APIs usually do this with something like:
```
context.extension.callOnClose(() => {
  if (context.extension.shutdownReason == "ADDON_UNINSTALL") {
    // ...
  }
});
```

Reply:
Sure, I might as well change that so I don't have to import AddonManager.